api/v1alpha2/common_types.go

```diff
@@ -126,3 +134,30 @@ type ObjectMeta struct {
 	// +patchStrategy=merge
 	OwnerReferences []metav1.OwnerReference `json:"ownerReferences,omitempty" patchStrategy:"merge" patchMergeKey:"uid" protobuf:"bytes,13,rep,name=ownerReferences"`
 }
+
+/*
+// InfrastructureCapabilities contains information for the node's address.
+type InfrastructureCapabilities struct {
```

Comment:
Alternative implementation using a struct.  The downside to this is we would have to build validation logic for each of these fields.

We might also consider a more nested structure, but I don't have any good ideas for that. (EG, power is a top-level struct, and different capabilities are nested under that).  I don't really have a strong preference.